src/internal/components/ThemeEditor.tsx

```diff
@@ -141,6 +145,7 @@ export const ThemeEditor = (props: ThemeEditorProps) => {
       const themeInitialHistory = {
         history: history,
         index: history.length - 1,
+        hash: themeSaveState.hash,
```

Comment:
We had to not use localStorage and set things up differently when we didn't have hash because we couldn't just look up by index as it wasn't unique. For example, in loadPuckInitialHistory we get the localStorage and look for the hash that matches saveState, but for themes we only had index so there's nothing to look up. Now that we have hash we should go back to using the same type of logic. Can you do that? Again, it's going to require changing the history object around I think.